### PR TITLE
Add support to set/change log retention period

### PR DESCRIPTION
**Pull request by nnbu**
Issue #, if available:

Description of changes:
This change adds the support to set/change the retention period for log-groups. By default, when log group is created, log retention period is set to 'never expire'. So, it is necessary to allow to change it.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: apis/v1alpha1/ack-generate-metadata.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
 ack_generate_info:
-  build_date: "2023-05-24T17:19:40Z"
-  build_hash: 6542e04cc357215c91df94e747e3d412fb26df97
-  go_version: go1.19.4
-  version: v0.26.1-3-g6542e04
-api_directory_checksum: 399f9b6fb5675139a83fc87a26271448ad5d9584
+  build_date: "2023-10-05T20:54:20Z"
+  build_hash: 7445de38211639a12e79992d154adab6e60f01fd
+  go_version: go1.20.3
+  version: v0.27.1-1-g7445de3
```
Comment (a-hilaly) [inline apis/v1alpha1/ack-generate-metadata.yaml:5]: This works with runtime 0.27 only

Comment (nnbu) [reply apis/v1alpha1/ack-generate-metadata.yaml:5]: Could you guide on how to move back to 0.27? Is it decided in code-generator repo - go.mod?

Comment (a-hilaly) [reply apis/v1alpha1/ack-generate-metadata.yaml]: I mean - using the code-generator 0.27.1 is correct here. You just need to pin to the latest ACK runtime in `go.mod` + do a `go mod tidy`

Comment (nnbu) [reply apis/v1alpha1/ack-generate-metadata.yaml:5]: Thanks @a-hilaly . I made the change. Most of the tests are passing. `/test cloudwatchlogs-metadata-file-test` failure seems to be unrelated though

**Comment by a-hilaly** [comment]
@nnbu can we change the service link in `metadata.yaml` to `https://docs.aws.amazon.com/cloudwatch/` ? we can merge right away

**Comment by nnbu** [comment]
> @nnbu can we change the service link in `metadata.yaml` to `https://docs.aws.amazon.com/cloudwatch/` ? we can merge right away

Thank you @a-hilaly . I made the change. Tests are passing now. Should I create a release 0.0.1 in next PR?

**Review by a-hilaly — APPROVED** [review]
Great stuff, thank you @nnbu !
/lgtm

**Comment by a-hilaly** [comment]
>Should I create a release 0.0.1 in next PR?

Yes please

**Pull request merged**